Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -254,7 +254,8 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
             // Priority queue orders by least first. So we compare by last access time to get the least last access time (i.e. oldest) first.
             var priorityQueue = new PriorityQueue<ContentHashWithLastAccessTimeAndReplicaCount>(
                 pageSize * 2,
-                Comparer<ContentHashWithLastAccessTimeAndReplicaCount>.Create((c1, c2) => c1.LastAccessTime.CompareTo(c2.LastAccessTime)));
+                // Assume that EffectiveLastAccessTime will always have a value.
+                Comparer<ContentHashWithLastAccessTimeAndReplicaCount>.Create((c1, c2) => c1.EffectiveLastAccessTime.Value.CompareTo(c2.EffectiveLastAccessTime.Value)));

```

Comment:
I'm not 100% sure that the assumption is true for all the cases.

Does it make sense to be null-safe here?

Reply:
In the function called below (GetEffectiveLastAccessTimes) we make sure that it is never null. I would rather something break if the assumption is false because it would mean something is wrong with the code

Reply:
Understood. Thanks.

---
In reply to: [293474376](https://github.com/microsoft/BuildXL/pull/456#discussion_r293474376) [](ancestors = 293474376)